### PR TITLE
Add ElasticSearchDSL package

### PR DESCRIPTION
**Pull request by shyim**
Project Page https://github.com/ongr-io/ElasticsearchDSL

### CODE REVIEW

File: packages/ElasticSearchDSL/src/Rector/MethodCall/MigrateFilterToQueryRector.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        if (! $this->isType($node, 'ONGR\ElasticsearchDSL\Search')) {
+            return null;
+        }
+
+        if (! $node->name instanceof Node\Identifier) {
+            return null;
+        }
+
+        if ($node->name->name !== 'addFilter') {
```
Comment (TomasVotruba) [inline packages/ElasticSearchDSL/src/Rector/MethodCall/MigrateFilterToQueryRector.php]: better `isName()`

```suggestion
        if (! $this->isName($node, 'addFilter')) {
```

And remove the part above:

```diff
-if (! $node->name instanceof Node\Identifier) {
-    return null;
-}
```

File: packages/ElasticSearchDSL/src/Rector/MethodCall/MigrateFilterToQueryRector.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            return null;
+        }
+
+        $node->name->name = 'addQuery';
+
+        $node->args[1] = new Node\Arg(new Node\Expr\ClassConstFetch(new Node\Name('\ONGR\ElasticsearchDSL\Query\Compound\BoolQuery'), new Node\Identifier('FILTER')));
```
Comment (TomasVotruba) [inline packages/ElasticSearchDSL/src/Rector/MethodCall/MigrateFilterToQueryRector.php]:
```suggestion
        $node->args[1] = new Node\Arg(new Node\Expr\ClassConstFetch(new Node\Name\FullyQualified('ONGR\ElasticsearchDSL\Query\Compound\BoolQuery'), new Node\Identifier('FILTER')));
```

File: packages/ElasticSearchDSL/tests/Rector/MethodCall/MigrateFilterToQueryRector/Fixture/fixture.php.inc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+class SomeClass
+{
+    public function run()
+    {
+        $search = new \ONGR\ElasticsearchDSL\Search();
+
+        $search->addFilter(
+            new \ONGR\ElasticsearchDSL\Query\TermsQuery('categoryIds', [1, 2])
+        );
```
Comment (TomasVotruba) [inline packages/ElasticSearchDSL/tests/Rector/MethodCall/MigrateFilterToQueryRector/Fixture/fixture.php.inc]: There should be test case skipping already present 2nd argument

**Comment by shyim** [comment]
Fixed all comments :)

**Comment by TomasVotruba** [comment]
Looks good :+1:  
Thanks for fast work

**Pull request merged**